skrub/_gap_encoder.py

```diff
@@ -456,7 +479,14 @@ def fit(self, X, y=None):
         self._input_name = sbd.name(X)
         self._random_state = check_random_state(self.random_state)
         is_null = sbd.to_numpy(sbd.is_null(X))
-        X = sbd.to_numpy(X)
+        result = self._fit_transform(sbd.to_numpy(X), is_null)
+
+        self.scaler_ = total_standard_deviation_scaler(result)
+        result /= self.scaler_
+
+        return self._post_process(X, result)
+
+    def _fit_transform(self, X, is_null):
```

Comment:
I am confused: why do we need a "_fit_transform" method in addition to the "fit_transform" method?

Having multiple nested methods has two drawbacks. First it can make make reading the codebase harder. Second, it creates an extra frame in the stack and the risk is that this frame holds references on objects, thus increasing the memory usage.